src/components/App/index.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from "react";
+import { ThemeProvider } from "@mui/material";
+import CssBaseline from "@mui/material/CssBaseline";
+import { useAuthState } from "react-firebase-hooks/auth";
+import { getAuth } from "firebase/auth";
+import { Router, navigate } from "@reach/router";
+import { IntlConfig, IntlProvider } from "react-intl";
+
+import Login from "../../views/Login";
+import theme from "../../theme";
+import { HomeView } from "../../views/Home";
+import { AuthorizedProps } from "./types";
+import { firebaseApp } from "../../index";
+
+type Translations = {
+  es: IntlConfig["messages"];
+  en: IntlConfig["messages"];
+};
+
+const translations: Translations = {
+  es: require("../../translations/es.json"),
+  en: require("../../translations/en.json"),
+};
+
+const navLang = navigator.language.split("-")[0];
+
+const lang = isTranslation(navLang) ? navLang : "en";
+
+function isTranslation(langugage?: string): langugage is keyof Translations {
+  if (!langugage) return false;
+  return Object.keys(translations).includes(langugage);
+}
+
+export const App = () => (
+  <IntlProvider locale={lang} messages={translations[lang]} defaultLocale="en">
```

Comment:
This is kind of a heavy typescript lift, so I thought I would just do it and explain it here.

You can find `IntlProvider` api [here](https://formatjs.io/docs/react-intl/components/#intlprovider). According to this API the `IntlProvider` has proptype `IntlConfig`. So using [Indexed Access Types](https://www.typescriptlang.org/docs/handbook/2/indexed-access-types.html), i can type the `translations` as a `key:IntlConfig["messages"]`.  That way anytime we take an object from that type, the compiler know it is the proper type for the `messages` prop.

Then we need to pull the language from the navigator. And we need to make sure that the language matches one of the keys of `translations` (in this case "es" or "en"). We can do that with [typeguards](https://basarat.gitbook.io/typescript/type-system/typeguard#user-defined-type-guards) using the `is` keyword. That function tells the compiler that `navLang` is a `key of Translation`. We need hardcode the keys in order for that to work, so hence why I created the `TranslationType`

Reply:
I wasn't aware of indexed access types or typeguards but makes perfect sense now. Thanks for the explainer.
This seems like a good solution